core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -2712,7 +2713,15 @@ public void createLanguageFunction(Session session, QualifiedObjectName name, La
         CatalogHandle catalogHandle = catalogMetadata.getCatalogHandle();
         ConnectorMetadata metadata = catalogMetadata.getMetadata(session);
 
-        metadata.createLanguageFunction(session.toConnectorSession(catalogHandle), name.asSchemaFunctionName(), function, replace);
+        SchemaFunctionName schemaFunctionName = name.asSchemaFunctionName();
+        if (catalogMetadata.getSecurityManagement() == SYSTEM) {
+            systemSecurityMetadata.functionCreated(
+                    session,
+                    new CatalogSchemaFunctionName(catalogHandle.getCatalogName().toString(), schemaFunctionName),
+                    new TrinoPrincipal(PrincipalType.USER, session.getUser()));
+        }
+
+        metadata.createLanguageFunction(session.toConnectorSession(catalogHandle), schemaFunctionName, function, replace);
```

Comment:
I mean these two statements should be switched, per convention at least:

```suggestion
        metadata.createLanguageFunction(session.toConnectorSession(catalogHandle), schemaFunctionName, function, replace);
        if (catalogMetadata.getSecurityManagement() == SYSTEM) {
            systemSecurityMetadata.functionCreated(
                    session,
                    new CatalogSchemaFunctionName(catalogHandle.getCatalogName().toString(), schemaFunctionName),
                    new TrinoPrincipal(PrincipalType.USER, session.getUser()));
        }
```